### PR TITLE
apps/testcase/audio : add scenario TCs for audio itc

### PR DESCRIPTION
**Pull request by arvinmittal**
filepath : TizenRT/apps/examples/testcase/ta_tc/audio/itc
file name : itc_audio_main.c

add scenario :
itc_audio_pcm_open_p_with_different_flag
itc_audio_pcm_open_p_reopen
itc_audio_pcm_open_n
itc_audio_pcm_close_n
itc_audio_pcm_is_ready_n_after_close
itc_audio_pcm_is_ready_n
itc_audio_pcm_prepare_n_after_close
itc_audio_pcm_open_by_name_n
itc_audio_pcm_get_config_n_after_close
itc_audio_pcm_get_config_n
itc_audio_pcm_get_channels_n_after_close
itc_audio_pcm_get_channels_n
itc_audio_pcm_get_rate_n_after_close
itc_audio_pcm_get_rate_n
itc_audio_pcm_get_format_n_after_close
itc_audio_pcm_get_format_n
itc_audio_pcm_get_file_descriptor_n_after_close
itc_audio_pcm_get_file_descriptor_n
itc_audio_pcm_get_error_n
itc_audio_pcm_get_buffer_size_n_after_close
itc_audio_pcm_get_buffer_size_n
itc_audio_pcm_get_subdevice_n_after_close
itc_audio_pcm_get_subdevice_n
itc_audio_pcm_frames_to_bytes_n_after_close
itc_audio_pcm_frames_to_bytes_n
itc_audio_pcm_bytes_to_frames_n_after_close
itc_audio_pcm_bytes_to_frames_n
itc_audio_pcm_format_to_bits_n
itc_audio_pcm_readi_n
itc_audio_pcm_writei_n

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check Result:
apps/examples/testcase/tc_common.h:37: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/tc_common.h:68: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/tc_common.h:80: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/tc_common.h:92: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/tc_common.h:104: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/tc_common.h:116: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
apps/examples/testcase/tc_common.h:128: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

**Comment by seinfra** [comment]
Code Rule Check OK.

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 1]
```diff
@@ -75,241 +64,755 @@ static void clean_all_data(int fd, char *buffer)
...
+static void itc_audio_pcm_open_p_reopen(void)
+{
+	struct pcm *ptr_pcm = NULL;
+	g_pcm = NULL;
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
+	// re-open pcm stream
+	ptr_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", ptr_pcm, NULL);
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: ptr_pcm will allocated, so it is not null, you should check validation with pcm_get_file_descriptor.

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c [hunk 2]
```diff
@@ -75,241 +64,755 @@ static void clean_all_data(int fd, char *buffer)
...
 * @postcondition    NA
 */
-static void itc_audio_pcm_is_ready_tc_p(void)
+static void itc_audio_pcm_is_ready_p(void)
 {
+	g_pcm = NULL;
 	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_NEQ("pcm_open", g_pcm, NULL);
 	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0);
 	TC_ASSERT_GT_CLEANUP("pcm_is_ready", pcm_is_ready(g_pcm), 0, pcm_close(g_pcm));
```
Comment (Taejun-Kwon) [inline apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: TC_ASSERT_EQ_CLEANUP is correct.

File: apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c (hunk 1 above)
Comment (arvinmittal) [reply apps/examples/testcase/ta_tc/audio/itc/itc_audio_main.c]: ptr_pcm is initiallised to NULL before. pcm_open on Failure can return NULL. Later ptr_pcm is also validated with pcm_get_file_descriptor

**Comment by arvinmittal** [comment]
Resolved merge conflicts and fix review comments in new commit. Also done ITCs name change.
 new commit PR : https://github.com/Samsung/TizenRT/pull/1083

**Pull request closed without merging**